### PR TITLE
[SeviceBus] Entity name validation when spawning handler from client

### PR DESCRIPTION
**Pull request by yunhaoling**

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py [hunk 1]
```diff
@@ -359,6 +380,13 @@ def get_subscription_receiver(self, topic_name, subscription_name, **kwargs):
 
         """
         # pylint: disable=protected-access
+
+        if self._entity_name and topic_name != self._entity_name:
```
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:384]: for sub receiver is it still just topic_name?  and not like topic_name/subscription_name?

Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:384]: (from OOB no, topic-only connstr as far as entity-specific goes.)

Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:384]: right, connection string only contains the topic name as the entity path

**Review by KieranBrantnerMagee — APPROVED** [review]

**Review by KieranBrantnerMagee — APPROVED** [review]
One of the old client tests had the service side exception expect for the client side check we just put in, was a simple/safe fix, reapproving/running++

**Pull request merged**